### DISCUSSION TITLE
Cannot access member xxx for protobuf msg

### DISCUSSION THREAD

**Discussion by AaronWaxson** [microsoft/pyright#4308] [General]
I used protobuf message in my project , but pyright can not access their members. Is there any solution for that?
I try to use `# type: ignore`, but too many lines with protobuf message. Maybe has a way to configure many types ignored?
Here's my example:
```proto
// geometry.proto
syntax = "proto2";

message Point2D {
  optional double x = 1 [default = nan];
  optional double y = 2 [default = nan];
}
```
```python
# test.py
from geometry_pb2 import Point2D

p1 = Point2D(1, 2)
print(p1.x, p1.y)
```
The x and y will be reported "Cannot access member x(y) for type Point2D, member x(y) is unknown".
By the way, I used pyright plugin in vscode.

**Comment by erictraut** [comment]
Moving this to a discussion topic because this isn't a bug in pyright.

**Reply by erictraut** [reply]
The Python code emitted by the `protoc` tool unfortunately doesn't contain type information. You need to create or generate type stubs if you want to use the generated Python modules in a typed project. There is a `protoc` plugin called [mypy-protobuf](https://github.com/nipunn1313/mypy-protobuf) that solves this problem. It generates ".pyi" files alongside the ".py" files.

**Reply by AaronWaxson** [reply]
Thanks a lot, I'll try it
